### DISCUSSION TITLE
Encountered an index error. Please ensure that all indices in 'edge_index' point to valid indices in the interval [0, 5916] (got interval [0, 6288])

### DISCUSSION THREAD

**Discussion by yaoysyao** [pyg-team/pytorch_geometric#7254] [Q&A] [answered]
When I used the create data set, nothing happened and I checked every data using data.validate(raise_on_error=True). Nothing happened. However, when I put GCNConv, I got the following exception: Exception:  Encountered an index error. Please ensure that all indices in 'edge_index' point to valid indices in the interval [0,  5916] (got interval [0, 6288]), I wonder what caused this error? Is it because the node indicated by my edge_index exceeds the actual node index?
Correct batch of databatch
DataBatch(x=[6289, 300], edge_index=[2, 21727], edge_attr=[21727, 1], y=[100], batch=[6289], ptr=[101])
Here is the databatch for the batch where the error occurred:
DataBatch(x=[5917, 300], edge_index=[2, 20481], edge_attr=[20481, 1], y=[100], batch=[5917], ptr=[101])

**Comment by wsad1** [comment]
> Is it because the node indicated by my edge_index exceeds the actual node index?

Yes thats probably what is happening. 

Hard to say why that's happening though. How did you batch your data. Feel free to share the part of your code that leads to this error, and the full stack trace.

**Reply by yaoysyao** [reply]
Thank you for your answer, my task is text classification, using the graph classification method, my data processing method is as follows: Construct each piece of data into a graph, each word in the data as the graph node, node starting from 0, set the batchsize to 100, each graph due to the different number of words in each piece of data, so the number of nodes in each graph is different, the number of sides is also different, do you have any suggestions or solutions to this? Or can you provide a way to process the data in batches? I looked at the source, but never saw how PYG could be used for the task of text sorting, thanks again

**Reply by yaoysyao** [reply, marked as answer]
I think I found the cause of the problem. I looked at the pyg source code and when I defined GCN, I used the method: GCNConv(nfeat, nhid,cached=True, normalize=True), I think it's cached=True because the model is using the previous edge_index, so I undid that, and I'm going to test it, I have tested 4 epochs and have found no problems so far
